File: bcs-common/pkg/otel/exporter/jaeger/collector.go
/*
 * Tencent is pleased to support the open source community by making Blueking Container Service available.
 * Copyright (C) 2019 THL A29 Limited, a Tencent company. All rights reserved.
 * Licensed under the MIT License (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 * http://opensource.org/licenses/MIT
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied. See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package jaeger

import (
	"net/http"

	"go.opentelemetry.io/otel/exporters/jaeger"
)

// CollectorEndpoint configs jaeger agent endpoint
type CollectorEndpoint struct {
	Endpoint         string                           `json:"endpoint,omitempty" usage:"endpoint for sending spans directly to a collector"`
	Username         string                           `json:"username,omitempty" usage:"username to be used for authentication with the collector endpoint"`
	Password         string                           `json:"password,omitempty" usage:"password to be used for authentication with the collector endpoint"`
	CollectorOptions []jaeger.CollectorEndpointOption `json:"-"`
}

// WithCollectorEndpoint defines the full URL to the Jaeger HTTP Thrift collector. This will
// use the following environment variables for configuration if no explicit option is provided:
//
// - OTEL_EXPORTER_JAEGER_ENDPOINT is the HTTP endpoint for sending spans directly to a collector.
// - OTEL_EXPORTER_JAEGER_USER is the username to be sent as authentication to the collector endpoint.
// - OTEL_EXPORTER_JAEGER_PASSWORD is the password to be sent as authentication to the collector endpoint.
//
// The passed options will take precedence over any environment variables.
// If neither values are provided for the endpoint, the default value of "http://localhost:14268/api/traces" will be used.
// If neither values are provided for the username or the password, they will not be set since there is no default.
func WithCollectorEndpoint(options ...jaeger.CollectorEndpointOption) jaeger.EndpointOption {
	return jaeger.WithCollectorEndpoint(options...)
}

// WithEndpoint is the URL for the Jaeger collector that spans are sent to.
// This option overrides any value set for the
// OTEL_EXPORTER_JAEGER_ENDPOINT environment variable.
// If this option is not passed and the environment variable is not set,
// "http://localhost:14268/api/traces" will be used by default.
func WithEndpoint(endpoint string) jaeger.CollectorEndpointOption {
	return jaeger.WithEndpoint(endpoint)
}

// WithUsername sets the username to be used in the authorization header sent for all requests to the collector.
// This option overrides any value set for the
// OTEL_EXPORTER_JAEGER_USER environment variable.
// If this option is not passed and the environment variable is not set, no username will be set.
func WithUsername(username string) jaeger.CollectorEndpointOption {
	return jaeger.WithUsername(username)
}

// WithPassword sets the password to be used in the authorization header sent for all requests to the collector.
// This option overrides any value set for the
// OTEL_EXPORTER_JAEGER_PASSWORD environment variable.
// If this option is not passed and the environment variable is not set, no password will be set.
func WithPassword(password string) jaeger.CollectorEndpointOption {
	return jaeger.WithPassword(password)
}

// WithHTTPClient sets the http client to be used to make request to the collector endpoint.
func WithHTTPClient(client *http.Client) jaeger.CollectorEndpointOption {
	return jaeger.WithHTTPClient(client)
}

File: bcs-ui/backend/components/databus/collector.py
# -*- coding: utf-8 -*-
"""
Tencent is pleased to support the open source community by making 蓝鲸智云PaaS平台社区版 (BlueKing PaaS Community
Edition) available.
Copyright (C) 2017-2021 THL A29 Limited, a Tencent company. All rights reserved.
Licensed under the MIT License (the "License"); you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://opensource.org/licenses/MIT

Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
specific language governing permissions and limitations under the License.
"""
from typing import Dict, List

from django.conf import settings

from backend.components.base import BaseHttpClient, BkApiClient, ComponentAuth, response_handler


class DataBusApiConfig:
    """databus API 配置"""

    def __init__(self, host: str):
        self.host = host

        # 日志采集规则 API 地址
        prefix_path = 'api/c/compapi/v2/bk_log'
        self.create_collect_config_url = f'{self.host}/{prefix_path}/create_bcs_collector'
        self.update_collect_config_url = f'{self.host}/{prefix_path}/update_bcs_collector/{{rule_id}}'
        self.list_collect_configs_url = f'{self.host}/{prefix_path}/list_bcs_collector?bcs_cluster_id={{cluster_id}}'
        self.delete_collect_config_url = f'{self.host}/{prefix_path}/delete_bcs_collector/{{rule_id}}'


class LogCollectorClient(BkApiClient):
    """日志采集规则管理 Client"""

    def __init__(self, auth: ComponentAuth):
        self._config = DataBusApiConfig(settings.COMPONENT_HOST)
        self._client = BaseHttpClient(auth.to_header_api_auth())

    @response_handler()
    def create_collect_config(self, config: Dict) -> Dict:
        return self._client.request_json('POST', self._config.create_collect_config_url, json=config)

    @response_handler()
    def update_collect_config(self, config_id: int, config: Dict) -> Dict:
        url = self._config.update_collect_config_url.format(rule_id=config_id)
        return self._client.request_json('POST', url, json=config)

    @response_handler(default=list)
    def list_collect_configs(self, cluster_id: str) -> List[Dict]:
        url = self._config.list_collect_configs_url.format(cluster_id=cluster_id)
        return self._client.request_json('GET', url)

    @response_handler()
    def delete_collect_config(self, config_id: int) -> Dict:
        url = self._config.delete_collect_config_url.format(rule_id=config_id)
        return self._client.request_json('DELETE', url)
